src/topology/builder.rs

```diff
@@ -76,7 +77,8 @@ pub async fn build_pieces(
         };
 
         let (output, control) = Fanout::new();
-        let pump = rx.map(Ok).forward(output).map_ok(|_| TaskOutput::Source);
+        let pump = Codecs::new(&source.codecs, rx);
```

Comment:
Since all events that are produced by sources directly arrive into the merged `Codec`s, sources don't need to handle decoding themselves anymore.

E.g. the `Socket` source should now emit the unprocessed variants `Event::Message` for UDP and `Event::Chunk` for TCP instead of producing an `Event::Log`.